src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/listeners/PrisonerDomainEventsListener.kt

```diff
@@ -22,16 +27,31 @@ class PrisonerDomainEventsListener(
 
   @JmsListener(destination = "prisoner", containerFactory = "hmppsQueueContainerFactoryProxy")
   fun onPrisonerChange(message: String) {
-    val sqsMessage: SQSMessage = objectMapper.readValue(message, SQSMessage::class.java)
+    val sqsMessage: SQSMessage = objectMapper.readValue(message)
     log.debug("Received message {}", sqsMessage.MessageId)
-    val (eventType, prisonerId) = objectMapper.readValue(sqsMessage.Message, HMPPSDomainEvent::class.java)
-    telemetryClient.trackEvent("prisoner-domain-event-received", mapOf("eventType" to eventType, "offenderNo" to prisonerId), null)
-    if (eventFeatureSwitch.isEnabled(eventType)) when (eventType) {
-      "prison-visit.booked" -> prisonVisitsService.createVisit(sqsMessage.Message.fromJson())
-      "prison-visit.cancelled" -> prisonVisitsService.cancelVisit(sqsMessage.Message.fromJson())
-      else -> log.info("Received a message I wasn't expecting {}", eventType)
-    } else {
-      log.warn("Feature switch is disabled for {}", eventType)
+    when (sqsMessage.Type) {
+      "Notification" -> {
+        val (eventType, prisonerId) = objectMapper.readValue<HMPPSDomainEvent>(sqsMessage.Message)
+        telemetryClient.trackEvent(
+          "prisoner-domain-event-received",
+          mapOf("eventType" to eventType, "offenderNo" to prisonerId),
+        )
+        if (eventFeatureSwitch.isEnabled(eventType)) when (eventType) {
+          "prison-visit.booked" -> prisonVisitsService.createVisit(objectMapper.readValue(sqsMessage.Message))
+          "prison-visit.cancelled" -> prisonVisitsService.cancelVisit(objectMapper.readValue(sqsMessage.Message))
```

Comment:
you you have just changed the `fromJson()` function below - a bit my readable than `objectMapper.readValue(sqsMessage.Message)` ?

Reply:
Do you mean you'd rather keep the fromJson() function?

Reply:
I mean could this be 
`"prison-visit.booked" -> prisonVisitsService.createVisit(sqsMessage.Message.fromJson())`

and add in a version of the `fromJson()` extension function we had before?